go/api/config_test.go

```diff
@@ -0,0 +1,129 @@
+// Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
```

Comment:
Note: I had to get rid of the tests folder and put this test file beside the implementation file. This seems to be the way to do things in Go. If the test file resides in a separate tests directory, it is not in the same package as the implementation, which means tests do not have access to any of the internal (private) symbols in the implementation. Looking at other Go projects they also have their test files beside their implementation files